src/queries/shops.js

```diff
@@ -0,0 +1,29 @@
+/**
+ * @name shops
+ * @summary Query the Shops collection for a list of shops.
+ *   If shopIds are provided, return the matching shops. Otherwise, return all the shops.
+ * @param {Object} context - an object containing the per-request state
+ * @param {Array} shopIds - optional array of shop IDs to return
+ * @returns {Promise<Object>} Products object Promise
+ */
+export default async function shops(context, shopIds) {
```

Comment:
It's better to do the signature as `(context, { shopIds } = {})` in case we want to add or remove other ways of filtering in the future.